client/src/js/modules/dc/views/autoprocattachments.js

```diff
@@ -20,23 +22,61 @@ define(['marionette',
+            // Passed into the column from the layout.
+            var iCatBaseUrl = this.column.escape('iCatUrl');
+            var isIndustry = this.column.escape('isIndustryProposal').toLowerCase() === 'true';
+            var isPurgedSession = this.column.escape('isPurgedSession').toLowerCase() === 'true';
+
+            // Files with "visit_persist_storage_dir_segment" (config.php) in their path are assumed to exist permanently and ignore the BLSESSION purged value
+            var isPersistentFile = persistentStorageSegment ? this.model.get('FILEPATH').includes(persistentStorageSegment) :  false;
+            
+            if (isPersistentFile == true | isPurgedSession == false) {
+
+                var baseDownloadUrl = `${app.apiurl}/download/${this.column.escape('urlRoot')}/attachments/${this.model.escape(this.column.get('idParam'))}`
+                
+                // Default behaviour for download btn
+                this.$el.html(`<a href="${baseDownloadUrl}/dl/2" class="button dl"><i class="fa fa-download"/> Download</a>`)
+
+                // Additional btns per filetype
+                switch (this.model.get('FILETYPE')) {
+                    case 'Log':
+                    case 'Logfile':
+                        this.$el.append(`<a class="vaplog button" href="${baseDownloadUrl}/dl/1"> <i class="fa fa-search"/> View </a>`);
+                        break;
+
+                    case 'Graph':
+                        this.$el.append('<a class="vapplot button" href="#"><i class="fa fa-line-chart"/> View</a>');
+                        break;
+
+                    case 'Result':
+                        if (app.options.get('ccp4_cloud_upload_url')) {
+                            this.$el.append('<a class="vapupload button" href="#"><i class="fa fa-cloud-upload"/> CCP4 Cloud</a>');
+                        }
+                        break;
+                
+                    default:
+                        break;
+                }
             }
-
-            if (app.options.get('ccp4_cloud_upload_url') && this.model.get('FILETYPE') == 'Result') {
-                this.$el.append('<a class="vapupload button" href="#"><i class="fa fa-cloud-upload"></i> CCP4 Cloud</a>')
+            else {
+                if (isPurgedSession == true) {
+                    // Append for ANY assets of purged industrial proposals
+                    if (isIndustry == true) { this.$el.html('<div>Deleted</div>') }
+                    else {
+                        // if !industrial Proposal && iCatURL is present in config, this file has been removed from store but may be available in iCat/Archive.
+                        iCatBaseUrl !== ""
+                            ? this.$el.html('<a class="button" href=' + iCatBaseUrl +' target="_blank"><i class="fa fa-external-link"/> iCat</a>') 
+                            : this.$el.html('<div>iCatURL missing</div>')
```

Comment:
Maybe just leave a blank space, for a hypothetical other facility using Synchweb but not iCat
```suggestion
                            : this.$el.html('')
```

Reply:
This would leave an empty field if the config is not set up. Might be useful to place something there to flag it to the user eg "config missing" / "Link missing" / a broken-link icon or something?

Reply:
True, but it's not really a user problem, it's a dev problem. You have a console warning that appears as well.

Reply:
Good point. I''l update and push the changes back up.